pkg/quota/v1/evaluator/core/services.go

```diff
@@ -128,9 +128,12 @@ func (p *serviceEvaluator) Usage(item runtime.Object) (corev1.ResourceList, erro
 		value := resource.NewQuantity(int64(ports), resource.DecimalSI)
 		result[corev1.ResourceServicesNodePorts] = *value
 	case corev1.ServiceTypeLoadBalancer:
-		// load balancer services need to count node ports and load balancers
-		value := resource.NewQuantity(int64(ports), resource.DecimalSI)
-		result[corev1.ResourceServicesNodePorts] = *value
+		// load balancer services need to count node ports unless creation of node ports
+		// is suspressed.
+		if svc.Spec.AllocateLoadBalancerNodePorts == nil || *svc.Spec.AllocateLoadBalancerNodePorts == true {
```

Comment:
Even this I feel may not be completely accurate and the quota value should be number of `svc.Spec.Ports` wiith nodePort > 0.

Reply:
You are right. When `nodePort:` is explicitly specified for a port it will be created regadless of the `allocateLoadBalancerNodePorts:` setting. Good catch :+1: